modules/spring-config/src/main/java/org/mule/config/spring/SpringRegistry.java

```diff
@@ -199,6 +203,24 @@ public Object lookupObject(String key)
         }
     }
 
+    private boolean shouldApplyPrototypeLifecycle(String key, Object object)
+    {
+        if (applicationContext.isSingleton(key))
+        {
+            return false;
+        }
+
+        if (object instanceof MessageProcessor)
```

Comment:
The reason is explained on the issue but I'll add a comment. Basically, prototype types other than these end up being used in a dynamic flow/pipeline which applies lifecycle, resulting on the MP's LC being applied twice.